files/image_config/logrotate/rsyslog.j2

```diff
@@ -123,5 +77,17 @@
                 /bin/kill -HUP $(cat /var/run/rsyslogd.pid)
             fi
         fi
+
+        # The clean up removes archived files till log partition has enough space
+        # So one time running during a logrotate is enough.
+        # Supposedly, the clean up happens on the firstaction which before the rotations.
+        # But there's a corner case, that on firstaction, no archived log files found(all files are origin logs, e.g. syslog, swss.rec)
+        # After rotated all log files are empty since log partition is full, then logrotate skips since it asserts notifempty and file size.
+        # On that condition, we need to clean up on postrotate
+        # Again, one time clean up during a logrotate round is enough, more running just add useless scanning.
+        # Hook and trigger clean up on rotation on syslog(last rotation log in the configuration), to avoid useless running on other log files
+        if [ $(echo $1 | grep -c "/var/log/syslog") -gt 0 ]; then
+{% include 'on_demand_archived_log_clean_up.j2' %}
```

Comment:
Is there some check that we need to invoke before doing the clean up a second time?